js/exporter/jspdf/export.js

```diff
@@ -68,11 +67,8 @@ export const Export = {
                 const styles = dataProvider.getStyles();
                 const dataRowsCount = dataProvider.getRowsCount();
 
-                if(tableWidth) {
-                    autoTableOptions.tableWidth = this.convertPixelsToPoints(tableWidth);
-                }
-
-                this.setColumnWidths(autoTableOptions, columns, keepColumnWidths);
+                const pdfColumnWidths = this.getDefaultPdfColumnWidths(autoTableOptions.tableWidth, dataProvider.getColumnsWidths());
+                this.updateColumnWidths(jsPDFDocument, autoTableOptions, pdfColumnWidths, keepColumnWidths);
```

Comment:
?
```
if(!keepColumnWidths) {
  const pdfColumnWidths = this.getDefaultPdfColumnWidths
```

Reply:
I think I understand your idea. keepColumnWidths === false then no need to calculate pdf column width.
```
if(keepColumnWidths) {
  const pdfColumnWidths = this.getDefaultPdfColumnWidths(...);
  this.updateColumnWidths(...);
}
```